performance_manager/lib/gtfs_static_table.py

```diff
@@ -0,0 +1,317 @@
+from typing import List, Optional
+
+import logging
+import pandas
+import numpy
+
+import sqlalchemy as sa
+from sqlalchemy.orm import sessionmaker
+
+from .s3_utils import read_parquet, file_list_from_s3
+from .postgres_utils import MetadataLog, get_unprocessed_files
+from .postgres_schema import (
+    StaticFeedInfo,
+    StaticTrips,
+    StaticRoutes,
+    StaticStops,
+    StaticStopTimes,
+    StaticCalendar,
+)
+from .gtfs_utils import start_time_to_seconds
+
+
+def get_static_parquet_paths(table_type: str, feed_info_path: str) -> List[str]:
+    """
+    get static table parquet files from FEED_INFO path
+    """
+    springboard_bucket = "mbta-ctd-dataplatform-dev-springboard"
+    static_prefix = feed_info_path.replace("FEED_INFO", table_type)
+    static_prefix = static_prefix.replace(f"{springboard_bucket}/", "")
+    return list(
+        uri for uri in file_list_from_s3(springboard_bucket, static_prefix)
+    )
+
+
+# pylint: disable=too-many-arguments
+def get_and_insert_static_tables(
+    table_name: str,
+    feed_info_path: str,
+    session: sa.orm.session.sessionmaker,
+    insert_table: str,
+    columns_to_pull: List[str],
+    int64_cols: Optional[List[str]] = None,
+    bool_cols: Optional[List[str]] = None,
+    time_to_seconds_cols: Optional[List[str]] = None,
+) -> None:
+    """
+    common static table processing and insert logic
+    """
+    paths_to_load = get_static_parquet_paths(table_name, feed_info_path)
+
+    data_table = read_parquet(paths_to_load, columns=columns_to_pull)
+
+    assert data_table.shape[0] > 0
+
+    data_table = data_table.drop_duplicates()
+
+    if int64_cols is not None:
+        for col in int64_cols:
+            data_table[col] = pandas.to_numeric(data_table[col]).astype("Int64")
+
+    if bool_cols is not None:
+        for col in bool_cols:
+            data_table[col] = numpy.where(
+                data_table[col] == 1, True, False
+            ).astype(numpy.bool8)
+
+    if time_to_seconds_cols is not None:
+        for col in time_to_seconds_cols:
+            data_table[col] = (
+                data_table[col].apply(start_time_to_seconds).astype("Int64")
+            )
+
+    with session.begin() as cursor:  # type: ignore
+        cursor.execute(
+            sa.insert(insert_table), data_table.to_dict(orient="records")
+        )
+
+
+# pylint: enable=too-many-arguments
+
+
+def process_feed_info(
+    feed_info_path: str, session: sa.orm.session.sessionmaker
+) -> None:
+    """
+    load feed_info table
+    """
+    table_name = "FEED_INFO"
+    insert_table = StaticFeedInfo.__table__
+    columns_to_pull = [
+        "feed_start_date",
+        "feed_end_date",
+        "feed_version",
+        "timestamp",
+    ]
+    int64_cols = [
+        "feed_start_date",
+        "feed_end_date",
+        "timestamp",
+    ]
```

Comment:
these things kind of feel like objects to me, but i don't think its crucial to change. if we were going to be putting more and more things into `get_and_insert_static_tables` it def would be, but i don't think thats the case.